lib/plug/ssl.ex

```diff
@@ -310,6 +313,12 @@ defmodule Plug.SSL do
     throw({:configure, message})
   end
 
+  defp keynew(list, key, position, value) do
+    if List.keymember?(list, key, position),
+      do: list,
+      else: List.keystore(list, key, position, value)
```

Comment:
Perhaps?

```suggestion
      else: [{key, value} | list]
```

Reply:
Added separately since it reduced the function to a one-liner.